application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubCommand.java

```diff
@@ -0,0 +1,127 @@
+package org.togetherjava.tjbot.commands.github;
+
+import net.dv8tion.jda.api.events.interaction.command.CommandAutoCompleteInteractionEvent;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import org.kohsuke.github.GHIssue;
+import org.kohsuke.github.GHIssueState;
+import org.togetherjava.tjbot.annotations.MethodsReturnNonnullByDefault;
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.utils.StringDistances;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+
+/**
+ * Slash command (/github) used to search for an issue in one of the repositories listed in the
+ * config. It even comes with auto completion!
+ */
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class GitHubCommand extends SlashCommandAdapter {
+    private static final long ONE_MINUTE_IN_MILLIS = 60_000L;
+
+    /**
+     * The command option name of this slash command (this is only there to make the linter happy
+     * lol)
+     */
+    private static final String COMMAND_OPTION_NAME = "title";
+
+    private final GitHubReference reference;
+
+    /**
+     * The last time the cache was updated (millis after Jan 1st 1970)
+     * <p>
+     * The cache updates every minute (see the ONE_MINUTE_IN_MILLIS constant)
+     */
+    private long lastCache;
```

Comment:
Time units have to be part of the variable name to avoid confusion `lastCacheMillis`. Also, u wouldnt need javadoc here if u would make the var name better, add the "update" part to it: `lastCacheUpdateMillis`. That said, please go with something high level instead, namely `Instant lastCacheUpdate;`.

Reply:
done